.github/workflows/ci.yml

```diff
@@ -189,6 +189,8 @@ jobs:
         run: python -m pytest modin/config/test
       - shell: bash -l {0}
         run: python -m pytest modin/test/test_envvar_catcher.py
+      - shell: bash -l {0}
+        run: python -m pytest modin/test/test_internals.py
```

Comment:
Do we need to add this `test_internals.py` in TeamCity testing scope?

Reply:
I think adding this file to testing is redundant. Both tests can be added into existing files.

Reply:
What file can you suggest? I would prefer these tests to be near.

Reply:
> Do we need to add this `test_internals.py` in TeamCity testing scope?

Tests from `test-internals` github action job doesn't include in TeamCity testing scope for now, so `test_internals.py` can also exist only in github actions CI.

Reply:
Do you plan to include this file in `TeamCity` testing scope?

Reply:
Based on the comment above - no, any objections?

Reply:
I haven't any objections for current state of this file.

Reply:
> What file can you suggest? I would prefer these tests to be near.

I would add them into `modin/pandas/test/dataframe/test_iter.py::test___repr`.

Reply:
But we actually doesn't test repr, it's just the trigger for check consistency of internal axes.

Reply:
I'm voting up to make separate tests to cover internal logic, but for this approach a have some objections:

1. These tests are engine-specific, the problem exists in described sense only for engines that inherits logic of partitioning from [`BasePandasFrame`](https://github.com/anmyachev/modin/blob/504fd910390255a72a4bd6fde092e43d42b5d8f4/modin/engines/base/frame/data.py#L27). For example, OmniSci backend have different engine and do not inherit that logic, so these tests don't make sense for OmniSci engine. My suggestion is to explicitly mark that we're testing only limited scope of engines here, maybe we can rename the path to that file like: `modin/test/engines/pandas/test_internals.py`

2. We should not use high-level abstractions like `DataFrame` or `Series` and use query compilers or `ModinFrame` instead.  I'm okay with creating frames with high-level API instruments, but I think that since we can extract from created frames `df._query_compiler` and `df._query_compiler._modin_frame` we should use them to recreate bad test case. Also, I think we should use some more advanced logic of checking that everything is okay instead of just `repr`

Reply:
Moved `test_internals.py` into `modin/test/backend/pandas/`.

As for the second statement, this applies only indirectly to this pull request. Create an appropriate issue for this.

Reply:
I like @dchigarev's suggestion. Regarding 1, [OmnisciOnRayFrame](https://github.com/modin-project/modin/blob/master/modin/experimental/engines/omnisci_on_ray/frame/data.py#L52) is inherited from `BasePandasFrame` as well. So, I would not divide location into `engines` folder for now and just keep location `modin/test/test_internals.py` for these tests. As expanding tests we will be able to divide them into appropriate locations. Regarding 2, I am okay with this. Also, we will need to create an appropriate issue for moving current tests related to internal errors into new location.

Reply:
I suggest leaving the folder for tests as it is, because one way or another we will have to change the folder for these tests in the future, so let's think about it when we decide exactly how it should look.

So we will quickly add this fix to the master, which users has been waiting for a long time.

Reply:
Okay.

Reply:
One more comment. Why didn't you include tests into [here](https://github.com/modin-project/modin/blob/master/.github/workflows/push.yml)?